tests/compas/robots/test_configuration.py

```diff
@@ -135,6 +136,43 @@ def test_fixed_length_list():
     assert len(fll) == 3
 
 
+def test_fixed_length_list_copy():
+    fll = FixedLengthList([[1, 2, 3] for _ in range(3)])
+    fll_copy = copy.copy(fll)
+    assert fll_copy[0][0] == 1
+    fll_copy[0][0] = 0
+    assert fll_copy[0][0] == 0
+    assert fll[0][0] == 0
+
+
+def test_fixed_length_list_deepcopy():
+    fll = FixedLengthList([[1, 2, 3] for _ in range(3)])
+    fll_copy = copy.deepcopy(fll)
+    assert fll_copy[0][0] == 1
+    fll_copy[0][0] = 0
+    assert fll_copy[0][0] == 0
+    assert fll[0][0] == 1
+
+
+def test_configuration_copy():
+    c = Configuration([1, 2, 3], [0, 0, 0])
+    c_copy = copy.copy(c)
+    assert c_copy.joint_values[0] == 1
+    c_copy.joint_values[0] = 0
+    assert c_copy.joint_values[0] == 0
+    assert c.joint_values[0] == 0
```

Comment:
tldr: Ironpython is really annoying, so I will delete this test.

Ironpython uses `copy` rather than `__copy__` and `Configuration.copy` is inherited from `Data.copy` which uses `deepcopy` and the 'data' setters and getters.  Something very mysterious happens in all the getting and setting and using other properties which results in a new `FixedLengthList` being created, rather than the expected behavior of just passing the reference.  Compare (with ipy) the output of 
```
class Thing(Data):
    def __init__(self, some_list):
        some_list = FixedLengthList(some_list or [])
        self._some_list = some_list
        
    @property
    def some_list(self):
        return self._some_list
        
    @some_list.setter
    def some_list(self, values):
        self._some_list = FixedLengthList(values)
        
    #@property
    def data(self):
        return {'some_list': self.some_list}
        
    #@data.setter
    #def data(self, data):
    #    self._some_list = FixedLengthList(data.get('some_list') or [])

t = Thing([1,2,3])
t_new = copy.copy(t)
t_new.some_list[0] = 0
print(t.some_list)
print(t_new.some_list)
``` 
with the same code but with the comments removed.  I don't understand why the commented version does anything sensical.  Why does `cls.from_data(deepcopy(self.data))` in `compas.data.Data.copy` make sense when `self.data` is a method?